console/v1/types_console_notification.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Extension for configuring openshift web console notifications.
+type ConsoleNotification struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleNotificationSpec `json:"spec"`
+}
+
+type ConsoleNotificationSpec struct {
+	// Text of the notification
+	Text string `json:"text"`
+	// The location of the notification
+	// +optional
+	Location NotificationLocationSelector `json:"location"`
+	// Object that holds notification link details
+	// +optional
+	Link ConsoleLink `json:"link"`
+	// The color of the text for the notification
+	// +optional
+	Color string `json:"color,omitempty"`
+	// The color of the background for the notification
+	// +optional
+	BackgroundColor string `json:"backgroundColor,omitempty"`
+}
+
+type NotificationLocationSelector string
```

Comment:
We should be consistent with our names for these types (`ConsoleLocationSelector` vs `NotificationLocationSelector`)

Reply:
I'll update to `ConsoleNotificationLocationSelector`, 35 chars ftw!

Reply:
Do we need the `Selector` on the end? I think just `ConsoleNotificationLocation` and `ConsoleLinkLocation` work

Reply:
Works for me.